module/VuFind/src/VuFind/RecordDataFormatter/Specs/DefaultRecord.php

```diff
@@ -0,0 +1,396 @@
+<?php
+
+/**
+ * DefaultRecord RecordDataFormatter specs.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  RecordDataFormatter
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:architecture:record_data_formatter
+ * Wiki
+ */
+
+namespace VuFind\RecordDataFormatter\Specs;
+
+use VuFind\View\Helper\Root\RecordDataFormatter\SpecBuilder;
+use VuFind\View\Helper\Root\SchemaOrg;
+
+use function count;
+
+/**
+ * DefaultRecord RecordDataFormatter specs.
+ *
+ * @category VuFind
+ * @package  RecordDataFormatter
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:architecture:record_data_formatter
+ * Wiki
+ */
+class DefaultRecord extends AbstractBase
+{
+    /**
+     * The order in which groups of authors are displayed.
+     *
+     * The dictionary keys here correspond to the dictionary keys in the $labels
+     * array in getAuthorFunction()
+     *
+     * @var array<string, int>
+     */
+    protected array $authorOrder = ['primary' => 1, 'corporate' => 2, 'secondary' => 3];
+
+    /**
+     * Constructor
+     *
+     * @param ?SchemaOrg $schemaOrgHelper schema.org helper
+     * @param array      $config          Config
+     */
+    public function __construct(protected ?SchemaOrg $schemaOrgHelper, array $config)
+    {
+        $this->config = $config;
+        $this->init();
```

Comment:
Would it make sense to have the abstract base class include a constructor that defines the config property with property promotion and calls init? Then you could just call `parent::__construct($config);` here. That would also help justify including the `init` method as an abstract in the parent class, since then it would actually get called there. (Alternatively, if you prefer not to refactor constructors, maybe it's not necessary to have that abstract method at all and leave it up to the author of the subclass where/when/how to perform initialization).